### PR TITLE
Add rock5a nvme boot support

### PR DESCRIPTION
**Pull request by amazingfate**

### CODE REVIEW

**Comment by RadxaStephen** [comment]
We cannot enable spi nor flash and eMMC at the same time.
See this commit.
https://github.com/radxa/u-boot/commit/6cb48b3786a3430afbb3219d6170ee9fe0878599

By default we want eMMC bootimg.
Can you disable nodes sfc and spi_nor?

Thanks.

**Comment by amazingfate** [comment]
I forgot that I've disabled emmc in device tree when tesing. Emmc and spi flash are both installed as a removable module so they can't get installed to board at the same time. Can the uboot driver itself auto detect if there is emmc or spi flash installed on board?

**Comment by RadxaStephen** [comment]
You can disable nodes sfc and spi_nor and keep nvme related nodes.
And we can support

* eMMC
* eMMC + NVME
* TF
* TF + NVME

For SPI + nvme and SPI + USB, We need to add another commit to enable sfc and spi_nor nodes and disable sdhci.

**Comment by RadxaStephen** [comment]
> I forgot that I've disabled emmc in device tree when tesing. Emmc and spi flash are both installed as a removable module so they can't get installed to board at the same time. Can the uboot driver itself auto detect if there is emmc or spi flash installed on board?

I think when you enable spi and emmc at the same, pinctrl driver would show error.

**Comment by amazingfate** [comment]
Force pushed, now it only has pci related changes.

**Comment by amazingfate** [comment]
@RadxaStephen @RadxaYuntian Refering to the implementation from orangepi5 which distinguish pcie and sata with two uboot device trees: https://github.com/orangepi-xunlong/u-boot-orangepi/commit/cdbf9851cdedffc9f0abda395da72d5292cc31e9,
I add one more defconfig and devicetree for spi flash.

File: arch/arm/dts/rk3588s-rock-5a-spi.dts
Comment (RadxaYuntian) [inline arch/arm/dts/rk3588s-rock-5a-spi.dts:1]: `#include "rk3588s-rock-5a.dts"` then adjust the eMMC/SFC node?

**Comment by amazingfate** [comment]
Build with armbian build framework seems okay: https://paste.next.armbian.com/eyutetofiv

**Comment by amazingfate** [comment]
Tested with spi flash module + nvme ssd and emmc, armbian is booting fine.

**Comment by RadxaStephen** [comment]
Thanks. I tested eMMC + NVME. It works.

And would you mind changing your git message?

Here are some rules:

* The commit message summary: no longer than 50 characters.
* The commit message description: limited to 72 characters characters for one line.
* Add Signed-off-by.

**Comment by RadxaStephen** [comment]
Great. Thanks.

**Comment by stevepringle** [comment]
Hi, is there any timescale for adding usb3.0 boot support to this?

**Pull request merged**